prisma/schema.prisma

```diff
@@ -38,6 +55,17 @@ model Gift {
     user      User?    @relation(fields: [userUUID], references: [uuid])
 }
 
+model Anniversary {
+    id        Int      @id @default(autoincrement())
+    uuid      String   @unique @default(uuid())
+    name      String
+    date      DateTime @db.Date
+    userUUID  String
+    personId  Person   @relation(fields: [userUUID], references: [userUUID], onDelete: Cascade)
+    createdAt DateTime @default(now())
+    updatedAt DateTime @updatedAt
+}
```

Comment:
Samat kentät mitä Issuessa. Lisäsin myös `userUUID`n, jota käytetään hyödyksi `personId`ssä kun etsitään relaatio `Person`-modelista.

Tässä itseasiassa lienee parempi käyttää `Person`-modelin `uuid`:tä eikä `User`-modelin `uuid`:tä? Koska kyse on `Person`:sta ei `User`:sta

`userUUID` ei ole `@unique`, se kannattanee olla? Toisaalta samalla persoonalla voi olla useampi merkkipäivä

Toinen mikä mietityttää on tuo `onDelete: Cascade`. Koska onko tämä `1-n` vai `1-1`? :smile: Rivillä 44 on asetettu seuraavasti `Anniversary   Anniversary[]`

Reply:
> Tässä itseasiassa lienee parempi käyttää `Person`-modelin `uuid`:tä eikä `User`-modelin `uuid`:tä? Koska kyse on `Person`:sta ei `User`:sta

Jep just näin, eli kannattaa uudelleennimetä `userUUID`-> `personUUID`

> `userUUID` ei ole `@unique`, se kannattanee olla? Toisaalta samalla persoonalla voi olla useampi merkkipäivä

Ei voi olla `@unique` koska samalla Personilla voi olla useampi merkkipäivä.

> Toinen mikä mietityttää on tuo `onDelete: Cascade`. Koska onko tämä `1-n` vai `1-1`? 😄 Rivillä 44 on asetettu seuraavasti `Anniversary Anniversary[]`

Tämä on 1-n relaatio jos yllä mainittua `@unique` ei aseteta. Se onko kyseessä 1-n vai 1-1 ei vaikuta tuohon `onDelete: Cascade` komentoon, sillä kummassakin tapauksessa kaikki Anniversaryt poistetaan jos Person poistetaan. Myöskään `Anniversary   Anniversary[]` ei vaikuta.